src/exposurelog/routers/delete_message.py

```diff
@@ -11,33 +9,32 @@
 from ..message import Message
 from ..shared_state import SharedState, get_shared_state
 
-router = fastapi.APIRouter(prefix="/exposurelog")
+router = fastapi.APIRouter()
 
 
-@router.post("/delete_message/", response_model=typing.List[Message])
-async def delete_messages(
-    ids: typing.List[int] = fastapi.Form(
-        ..., title="IDs of messages to delete"
-    ),
-    site_id: str = fastapi.Form(..., title="Site ID of messages to delete"),
+@router.delete("/messages/{site_id}_{id}", response_model=Message)
+async def delete_message(
+    id: int,
+    site_id: str,
     state: SharedState = fastapi.Depends(get_shared_state),
-) -> typing.List[Message]:
-    """Mark one or more messages as deleted.
+) -> Message:
+    """Delete a message by marking it invalid.
 
-    Do this by setting ``is_valid`` False and updating
-    ``date_is_valid_changed``. Return the deleted messages.
+    Set ``is_valid`` False and update ``date_is_valid_changed``.
+    Return the deleted message.
```

Comment:
The DELETE method doesn't return the deleted resource. Normally I'd recommend returning a 204 with an empty body. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/DELETE

Reply:
Deleted messages are just marked as invalid. I thought it would be potentially useful to return the deleted resource to show that the is_valid flag was set False and the related datestamp was updated. Is it actively harmful to return this information?

Reply:
Also DELETE is not quite idempotent -- the date_is_valid_set is always updated. I'm trying to fix that, but am not sure of how to do it with SQLAlchemy and still tell the difference between "was already deleted" and "record does not exist". In other words this is not good enough:
```
connection.execute(exposurelog_db.table
    .update()
    .where(sa.sql.and_(exposurelog_db.table.c.id == id, exposurelog_db.table.c.is_valid)
    .values(is_valid=False, date_is_valid_changed=current_time)
```

Reply:
> Deleted messages are just marked as invalid. I thought it would be useful to return the deleted resource to show that the is_valid flag was set False and the related timestamp was updated. Does that change your suggestion or would you still prefer I return 204 and an empty body?

It sounds to me like `is_valid` is more like an internal detail. I'd still just return `204` and an empty body.

Reply:
OK. Done.